src/parser/contract-schemas.client.ts

```diff
@@ -0,0 +1,127 @@
+import { CasperServiceByJsonRPC, decodeBase16 } from 'casper-js-sdk';
+
+import {
+  EVENTS_NAMED_KEY,
+  EVENTS_SCHEMA_NAMED_KEY,
+  parseSchemasFromBytes,
+  Schemas,
+} from './schema';
+
+export interface ContractSchemas {
+  schemas: Schemas;
+  contractHash: Uint8Array;
+  contractPackageHash: Uint8Array;
+  eventsSchemaUref: string;
+  eventsUref: string;
+}
+
+export class ContractSchemasClient extends CasperServiceByJsonRPC {
+  constructor(nodeUrl: string) {
+    super(nodeUrl);
+  }
+
+  public async fetchContractSchemasBytes(
+    contractHash: string,
+    stateRootHash?: string,
+  ): Promise<Uint8Array> {
+    if (!stateRootHash) {
+      stateRootHash = await this.getStateRootHash();
+    }
+
+    const contractData = (
+      await this.getBlockState(stateRootHash, `hash-${contractHash}`, [])
+    ).Contract;
+
+    if (!contractData) {
+      throw new Error('contract data not found');
+    }
+
+    const eventsSchema = contractData.namedKeys.find(
+      el => el.name === EVENTS_SCHEMA_NAMED_KEY,
+    );
+    if (!eventsSchema) {
+      throw new Error(
+        `'${EVENTS_SCHEMA_NAMED_KEY}' uref not found for contract '${contractHash}'`,
+      );
+    }
+
+    const schemaResponse = await this.client.request({
+      method: 'state_get_item',
+      params: {
+        state_root_hash: stateRootHash,
+        key: eventsSchema.key,
+        path: [],
```

Comment:
Instead of fetching contract and then it's uref it's possible to fetch the uref value in one request by providing the contract as the key and named key as the path. Unless I imagining it :)